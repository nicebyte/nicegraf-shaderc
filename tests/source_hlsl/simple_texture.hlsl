// T: simple_texture vs:VSMain ps:PSMain meta:Aaa=Bbb meta:x=567

#include "inc/triangle.hlsl"

[[vk::binding(1, 0)]] uniform Texture2D tex;
[[vk::binding(2, 0)]] uniform sampler samp;

float4 PSMain(Triangle_PSInput ps_in) : SV_TARGET {
  return tex.Sample(samp, ps_in.texcoord);
}

Triangle_PSInput VSMain(uint vid : SV_VertexID) {
  return Triangle(vid, 1.0);
}