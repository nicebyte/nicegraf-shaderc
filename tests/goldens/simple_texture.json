{
"header": {
  "magic_number": 3735928559,
  "header_size": 36,
  "version_maj": 0,
  "version_min": 1,
  "entrypoints_offset": 36,
  "pipeline_layout_offset": 80,
  "image_to_cis_map_offset": 112,
  "sampler_to_cis_map_offset": 132,
  "user_metadata_offset": 152
},
"entrypoints": { 
  "vertex": "VSMain",
  "fragment": "PSMain"
},
"pipeline_layout": {
  "descriptor_sets": [
    {
      "set": 0,
      "descriptors": [
        {
          "binding": 1,
          "type": "IMAGE",
          "stage_vis": 2
        },
        {
          "binding": 2,
          "type": "SAMPLER",
          "stage_vis": 2
        }
      ]
    }
  ]
},
"image_to_cis_map": {
  "entries": [
    {
      "entry": 0,
      "separate_set_id": 0,
      "separate_binding_id": 1,
      "combined_ids": [0]
    }
  ]
},
"sampler_to_cis_map": {
  "entries": [
    {
      "entry": 0,
      "separate_set_id": 0,
      "separate_binding_id": 2,
      "combined_ids": [0]
    }
  ]
},
"user_metadata": {
  "Aaa": "Bbb",
  "x": "567"
}
}
